### PR TITLE
IG changed there data structure.  - post_page[0]["shortcode_media"]

### PR DESCRIPTION
**Pull request by roscoe72**
## Description
IG changed there data structure.  - post_page[0]["shortcode_media"]

Fixes # (issue)
- Primary error: 'post_page[0]["shortcode_media"] KeyError: 0'
- Added more try/except duo IG not always include data in post_data is item has no data

ToDo: 
- Not all the edge_comments are always in de comments dict. 
- Remove rewrite query by modifying call to getMediaData functions in other files (util..py)

## How Has This Been Tested?
Tested on ubuntu20.04 and debian11 clients. 

## Checklist:
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have checked my code and corrected any misspellings
- [x] I have performed a self-review of my own code
- [x] My code follows the style guidelines of this project, `black -t py34`
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by Chessy11** [comment]
window._sharedData.entry_data.PostPage[0].media is undefined

**Comment by ElBartu** [comment]
not fix here

**Comment by ElBartu** [comment]
> window._sharedData.entry_data.PostPage[0].media is undefined

same

**Comment by Ticunas** [comment]
Facing this problem:
```
Traceback (most recent call last):
  File "py_script.py", line 149, in <module>
    bot.like_by_tags(like_tags,use_random_tags=True,randomize=True, amount=random.randint(5,15))
  File "/usr/local/lib/python3.7/site-packages/instapy/instapy.py", line 1990, in like_by_tags
    self.logger,
  File "/usr/local/lib/python3.7/site-packages/instapy/like_util.py", line 626, in check_link
    media = post_page["items"][0]
KeyError: 'items'
```

**Comment by Chessy11** [comment]
Does this commit works? Or am i missing something..

File: instapy/like_util.py [hunk 1]
```diff
@@ -589,37 +590,44 @@ def check_link(
         return True, None, None, "Unavailable Page", "Failure"
 
     # Gets the description of the post's link and checks for the dont_like tags
-    graphql = "graphql" in post_page
+    #DEF: 20jan
+    graphql = "items" in post_page
```
Comment (Ticunas) [inline instapy/like_util.py]: There is a case that still named as "graphql"
`post_page: {'graphql': {'shortcode_media': {'__typename': 'GraphImage', 'id': '2759238706931426441', 'shortcode': 'CZKyPwhJ-SJ', 'dimensions': {'height': 1080, 'width': 1080}, 'gating_info': None, 'fact_check_overall_rating': None, 'fact_check_information': None, 'sensitivity_friction_info': None, 'sharing_friction_info': {'should_have_sharing_friction': False, 'bloks_app_url': None}, 'media_overlay_info': None, 'media_preview': 'ACoqt+QB1A56Y6jjqf6YoMKgZ4/L/wCtVX7dKeP6DpVeVnmJYHA9Mtjp7EVXOlv+Rha/l95caND0IHbj1PpWfKxV9q5wMc//AFveoWCoMsef7o3ZP4lsY/X2qAyIRnbz7k/41nvrp9xoo6FjzSvUnmofNf1P51CXyMgYppkNKwWNEsUClcgkEnjOf89Ki3sTgHOecbf/AK9WxZIB5jMQByeCR+QqJBG22NG3nrkjAyOB15wOTj1NO6sO1xof5duMlgdwIzj3HfPvSQJG2dwyAD1zyT0/X8qdtlkdfKXYz5yc8ccE57D/APVipYbWXzfJdiCFJAB69Pw6mhbF2SMhhjI9KZVpFU7g2c8gYx17Zz70z7O/p+oqiTb3mUFnBCE/KvsO5+vXmiMREhlUZJwDx+dK3K881PGigjAHHtWKtezRs720/ImcQx/Meq8dce/QVnfa8vvhXjBHG4nqD0GcZ9+uKz7/AP1hNP00kTD3Bz+VayslYzjd633HGBmbcFA9vmxn8cYqyGmHGxfyb/GrknDcVaFY81+n5/5l28/wP//Z', 'display_url': 'https://instagram.fnat5-1.fna.fbcdn.net/v/t51.2885-15/e35/272632406_119773497240925_3029844279487032232_n.webp.jpg?_nc_ht=instagram.fnat5-1.fna.fbcdn.net&_nc_cat=101&_nc_ohc=Z6OgGKog2gYAX_mq9hy&edm=AABBvjUBAAAA&ccb=7-4&oh=00_AT8QpVIKyrDFbnA7PFxFMnhb3moedXP7wLcGXDYzWWT2fg&oe=61F6CE31&_nc_sid=83d603', 'display_resources': [{'src': 'https://instagram.fnat5-1.fna.fbcdn.net/v/t51.2885-15/sh0.08/e35/s640x640/272632406_119773497240925_3029844279487032232_n.webp.jpg?_nc_ht=instagram.fnat5-1.fna.fbcdn.net&_nc_cat=101&_nc_ohc=Z6OgGKog2gYAX_mq9hy&edm=AABBvjUBAAAA&ccb=7-4&oh=00_AT-Mvr5tzZOXfpwUBoE9MlccGViL803fs3HXNbljmcZtJQ&oe=61F78FD0&_nc_sid=83d603', 'config_width': 640, 'config_height': 640}, {'src': 'https://instagram.fnat5-1.fna.fbcdn.net/v/t51.2885-15/sh0.08/e35/s750x750/272632406_119773497240925_3029844279487032232_n.webp.jpg?_nc_ht=instagram.fnat5-1.fna.fbcdn.net&_nc_cat=101&_nc_ohc=Z6OgGKog2gYAX_mq9hy&edm=AABBvjUBAAAA&ccb=7-4&oh=00_AT_KFAqYCzjuce0Wz2rNVE9hiljuj3JdFngEE2tlJCVRTg&oe=61F772F9&_nc_sid=83d603', 'config_width': 750, 'config_height': 750}, {'src': 'https://instagram.fnat5-1.fna.fbcdn.net/v/t51.2885-15/e35/272632406_119773497240925_3029844279487032232_n.webp.jpg?_nc_ht=instagram.fnat5-1.fna.fbcdn.net&_nc_cat=101&_nc_ohc=Z6OgGKog2gYAX_mq9hy&edm=AABBvjUBAAAA&ccb=7-4&oh=00_AT8QpVIKyrDFbnA7PFxFMnhb3moedXP7wLcGXDYzWWT2fg&oe=61F6CE31&_nc_sid=83d603', 'config_width': 1080, 'config_height': 1080}], 'accessibility_caption': "Photo by Fisgou Postou 🎣 on January 25, 2022. May be an image of outdoors and text that says 'danilogasqui @pesqueironazarepedro @fisganois @fisgouposto'.", 'is_video': False, 'tracking_token': 'eyJ2ZXJzaW9uIjo1LCJwYXlsb2FkIjp7ImlzX2FuYWx5dGljc190cmFja2VkIjp0cnVlLCJ1dWlkIjoiNzZiN2NlZTIzNzQ2NDc1ZGIwNjI3ODdjNjhiYTEwYzAyNzU5MjM4NzA2OTMxNDI2NDQxIiwic2VydmVyX3Rva2VuIjoiMTY0MzE0Njk3NzU3N3wyNzU5MjM4NzA2OTMxNDI2NDQxfDUwODQwODM5MzcxfGM2ZjliYmRhNDEzOGRhZjMyMTFkMmVhN2NiYWVlMzRjZTUwMWIzZjUyMmM1ZGE4NDAxMTc2MjU0NjU4OGI1OWQifSwic2lnbmF0dXJlIjoiIn0=', 'upcoming_event': None, 'edge_media_to_tagged_user': {'edges': []}, 'edge_media_to_caption': {'edges': [{'node': {'text': '💯🎣🎣🎣\nPescador @danilogasqui \n\n#pescarias #pescaria #pescaesportiva #pesca #fishing #pescariatop #pescariaesportiva #fish #pescador #pesqueiro #pescariasesportivasbrasil #pesqueesolte #pescar #fishtv #pescaesolte #tambacu #pescadores #tambaqui #amopescar #pescadoresportivo #tamba #pescando #tucunare #pescabrasil #vidadepescador #natureza #pescariaboa #tucunar #pesquesolte #fisgoupostou'}}]}, 'can_see_insights_as_brand': False, 'caption_is_edited': False, 'has_ranked_comments': False, 'like_and_view_counts_disabled': False, 'edge_media_to_parent_comment': {'count': 0, 'page_info': {'has_next_page': False, 'end_cursor': None}, 'edges': []}, 'edge_media_to_hoisted_comment': {'edges': []}, 'edge_media_preview_comment': {'count': 0, 'edges': []}, 'comments_disabled': False, 'commenting_disabled_for_viewer': False, 'taken_at_timestamp': 1643146907, 'edge_media_preview_like': {'count': 0, 'edges': []}, 'edge_media_to_sponsor_user': {'edges': []}, 'is_affiliate': False, 'is_paid_partnership': False, 'location': None, 'viewer_has_liked': False, 'viewer_has_saved': False, 'viewer_has_saved_to_collection': False, 'viewer_in_photo_of_you': False, 'viewer_can_reshare': True, 'owner': {'id': '5564080665', 'is_verified': False, 'profile_pic_url': 'https://instagram.fnat5-1.fna.fbcdn.net/v/t51.2885-19/s150x150/24274472_412714685813894_4416795340431163392_n.jpg?_nc_ht=instagram.fnat5-1.fna.fbcdn.net&_nc_cat=102&_nc_ohc=0pCMHxyDi-4AX9e2dCZ&tn=XUbt_HHdxrOJDnFD&edm=AABBvjUBAAAA&ccb=7-4&oh=00_AT_vBOlPqWtRA89F-snxT8Jo5s7wFvTQ9WXp2Hugrr2GtA&oe=61F85796&_nc_sid=83d603', 'username': 'fisgoupostou', 'blocked_by_viewer': False, 'restricted_by_viewer': False, 'followed_by_viewer': False, 'full_name': 'Fisgou Postou 🎣', 'has_blocked_viewer': False, 'is_embeds_disabled': False, 'is_private': False, 'is_unpublished': False, 'requested_by_viewer': False, 'pass_tiering_recommendation': True, 'edge_owner_to_timeline_media': {'count': 345}, 'edge_followed_by': {'count': 1649}}, 'is_ad': False, 'edge_web_media_to_related_media': {'edges': []}, 'coauthor_producers': [], 'edge_related_profiles': {'edges': []}}}}`

Comment (roscoe72) [reply instapy/like_util.py]: Hi. 
Strange. When did you got this error? And on which actions?  session.like_by_tags, session.interact_user_followers??

Comment (Goli777) [reply instapy/like_util.py]: Hey @ErwinKoekoek guess what!?
It backs again =(
like_by_tags raised:
likes_count = post_page["graphql"]["shortcode_media"]["edge_media_preview_like"][
KeyError: 'graphql'


When signed in:
```
{
  "items": [
    {
      "taken_at": 1643181905,
      "pk": 2759532287433784300,
      "id": "2759532287433784205_6863203924",
      "device_timestamp": 1643181752705398,
      "media_type": 1,
      "code": "CZL0_6qtjuN",
      "client_cache_key": "Mjc1OTUzMjI4NzQzMzc4NDIwNQ==.2",
      "filter_type": 0,
      "is_unified_video": false,
      "user": {
        "pk": 6863203924,
        "username": "zoutse",
        "full_name": "𝑍𝑜𝑢",
        "is_private": false,
        "profile_pic_url": "https://scontent-frt3-1.cdninstagram.com/v/t51.2885-19/s150x150/271216498_3131291847158684_2891101624475983622_n.jpg?_nc_ht=scontent-frt3-1.cdninstagram.com&_nc_cat=108&_nc_ohc=4tclY1nrruEAX8L4ByH&tn=6fyLvxCxES2bfKhG&edm=AABBvjUBAAAA&ccb=7-4&oh=00_AT_yhO8ZQsuaqohs0DdtxTnvFLKKImp3oR552LlmuHN2iA&oe=61F7CC97&_nc_sid=83d603",
        "profile_pic_id": "2743077748342142410_6863203924",
        "friendship_status": {
          "following": false,
          "outgoing_request": false,
          "is_bestie": false,
          "is_restricted": false,
          "is_feed_favorite": false
        },
```
When NOT signed in:
```
{
  "graphql": {
    "shortcode_media": {
      "__typename": "GraphImage",
      "id": "2759532287433784205",
      "shortcode": "CZL0_6qtjuN",
      "dimensions": {
        "height": 1350,
        "width": 1080
      },
      "gating_info": null,
      "fact_check_overall_rating": null,
      "fact_check_information": null,
      "sensitivity_friction_info": null,
      "sharing_friction_info": {
        "should_have_sharing_friction": false,
        "bloks_app_url": null
      },
```
Back again. =( 

I think there should be a “fork” in code to check what data retrieved

get about an hour ago

Comment (roscoe72) [reply instapy/like_util.py]: Hi. Yes i'm looking at it. And trying to reproduce. Most times the fixes are working ;-)

Comment (Goli777) [reply instapy/like_util.py]: @ErwinKoekoek it's on my local machine. So right now i got the same script running on a remote server. And it's ok

Comment (Goli777) [reply instapy/like_util.py]: But since it happened once, it definitely should be twice. 
So I guess it is necessary to check the income data.

Comment (Ticunas) [reply instapy/like_util.py]: > Hi. Strange. When did you got this error? And on which actions? session.like_by_tags, session.interact_user_followers??

Sorry, more information about this issue:

Traceback (most recent call last):
  File "py_script.py", line 149, in <module>
    bot.like_by_tags(like_tags,use_random_tags=True,randomize=True, amount=random.randint(5,15))
  File "/usr/local/lib/python3.7/site-packages/instapy/instapy.py", line 1990, in like_by_tags
    self.logger,
  File "/usr/local/lib/python3.7/site-packages/instapy/like_util.py", line 626, in check_link
    media = post_page["items"][0]
KeyError: 'items'

Comment (roscoe72) [reply instapy/like_util.py]: > @ErwinKoekoek it's on my local machine. So right now i got the same script running on a remote server. And it's ok

@Goli777 Do you also the full so-called 'When NOT signed in' dump for me? What i can't reproduce i can't fix...
if other also have also _this_ specific dump please share that dump. Need some examples to work with...

Comment (Goli777) [reply instapy/like_util.py]: idk how to dump it

**Comment by ErwinKoekoek** [comment]
> not fix here

What is the issue? Can you be more specific about the issue?
Best 
Erwin

File: instapy/like_util.py (hunk 1 above)
Comment (roscoe72) [reply instapy/like_util.py]: > idk how to dump it
> 
> 

Hi @Goli777 Please dump it here in the thread.

Comment (Goli777) [reply instapy/like_util.py]: @ErwinKoekoek, I mean what data I have to dump?

**Comment by ErwinKoekoek** [comment]
@all. 
Just made a small fix that hopefully catches the last issues...
Did some test runs on my ubuntu and debian machines. 
=== Update like_util.py with try, except ===
https://github.com/ErwinKoekoek/InstaPy/tree/master/instapy


If you had problems and they are now solved. Please drop a line here. Lets close and finish this PR!!!

File: instapy/like_util.py (hunk 1 above)
Comment (Goli777) [reply instapy/like_util.py]: https://drive.google.com/drive/folders/1rxzMpc50-eife5QAQ6IHb09pCc4ZAHqj?usp=sharing
jsons of both logged in to instagram and not logged in

Comment (Goli777) [reply instapy/like_util.py]: @ErwinKoekoek how to dm you Telegram maybe?

Comment (roscoe72) [reply instapy/like_util.py]: Just made a fix. I think we solved it finger crossed 🤞

Comment (Goli777) [reply instapy/like_util.py]: i've a question not about this issue but about using poxy. Can you please help me. DM in any messanger please!

Comment (roscoe72) [reply instapy/like_util.py]: > https://drive.google.com/drive/folders/1rxzMpc50-eife5QAQ6IHb09pCc4ZAHqj?usp=sharing jsons of both logged in to instagram and not logged in

I got them. Thanks! Tomorrow i will look in it and test it on my test servers. 
Cheers
Erwin

Comment (Ticunas) [reply instapy/like_util.py]: Wow! Thank you @ErwinKoekoek, is working!

Cheers

**Comment by rebergman** [comment]
@ErwinKoekoek You're fix worked for my issues with liking posts. However, commenting and following are not working. Any ideas here? Happy to add more details if needed.

**Comment by PhineasFisher** [comment]
> Same Error on Win 10 also I have done #6467 corrections also.
> 
> Please help!!!

#6467 did the fix for me.

1. Make sure to backup the following files:
- like_util.py
- util.py
- comment_util.py

On linux these files are stored in : /usr/local/lib/python3/dist-packages/instapy

2. Then remove the contents of the original files (like_util.py , util.py and comment.py)

3. simply copy-paste the entire like_util.py and util.py from the link below into your instapy instance wherever your util files are located.
https://github.com/InstaPy/InstaPy/pull/6467/commits/cad38a13a4f3ea2ba6283064120a1cdc6a7bff30

4. copy paste the whole comment_util.py content from the link below to the corresponding file of your instapy instance
https://github.com/InstaPy/InstaPy/pull/6467/commits/61a9d152305dcbd664d5c1fe56af4ac1f2a9c781

5. In all 3 files, simply delete or comment out (#) the red lines from the links above.

6. Instapy should be working again.

**Comment by pjwarner** [comment]
I pulled these from https://github.com/ErwinKoekoek/InstaPy/tree/master/instapy and it seems that it did fix the problem that made everything not work at all, but it does not seem to respect the percentages to like.

For instance i have 

session.set_do_like(enabled=True, percentage=50)
session.like_by_tags(["gaming", "twitch", "streamer", "streaming", "gamer"], amount=12)

And it went through 105 photos ( (12+9) * 5) which is correct
And it liked 103 photos (One was Inappropriate and one was an invalid user)

**Comment by ErwinKoekoek** [comment]
> I pulled these from https://github.com/ErwinKoekoek/InstaPy/tree/master/instapy and it seems that it did fix the problem that made everything not work at all, but it does not seem to respect the percentages to like.
> 
> For instance i have
> 
> session.set_do_like(enabled=True, percentage=50) session.like_by_tags(["gaming", "twitch", "streamer", "streaming", "gamer"], amount=12)
> 
> And it went through 105 photos ( (12+9) * 5) which is correct And it liked 103 photos (One was Inappropriate and one was an invalid user)
> 
> Should enabled=True just be True?
Hi Paul,
I think this is a new or other issue then the fixes for the IG data structure. It’s better to make a new issue if you have problems.

**Comment by workproduct** [comment]
have tried both https://github.com/ErwinKoekoek/InstaPy and latest update of instapy through pip update and both are still giving graph errors. have cleared out cache folder from user folder as well.

**Comment by hokum-ha** [comment]
Not working consistently. Fixed about 80% of cases but no can't identify if user is private, along with can't like some images.

**Comment by martini09617** [comment]
@ErwinKoekoek 
Could you tell me in depth how to tackle this error? I am no pro, thus please explain what to do and where to do it. 
I am working on Windows 10, Python 3 and via Firefox.

The quickstart script is not working at this moment.

**Comment by martini09617** [comment]
Path in windows: C:\Users\'YOURNAME'\AppData\Local\Programs\Python\Python310\Lib\site-packages\instapy

**Comment by ErwinKoekoek** [comment]
> @ErwinKoekoek Could you tell me in depth how to tackle this error? I am no pro, thus please explain what to do and where to do it. I am working on Windows 10, Python 3 and via Firefox.
> 
> The quickstart script is not working at this moment.

No sorry I’m not a windows guy. It’s seems to work for most of them. 
best
Erwin

**Comment by ErwinKoekoek** [comment]
> Not working consistently. Fixed about 80% of cases but now can't identify if user is private, along with can't like some images attached to graph queries.

please share more debug and settings so I can try to reproduce

**Comment by ErwinKoekoek** [comment]
> have tried both https://github.com/ErwinKoekoek/InstaPy and latest update of instapy through pip update and both are still giving graph errors. have cleared out cache folder from user folder as well.

Yes you need the latest version of instapy. And then you need to update the three updates files from my repo.
That’s it.

**Comment by martini09617** [comment]
@ErwinKoekoek 
Got it, only the "like_util.py" is still working properly. I am getting an error mentioned here. #6233

**Comment by Ticunas** [comment]
> @ErwinKoekoek Got it, only the "like_util.py" is still working properly. I am getting an error mentioned here. #6233
> 
> More specific, the bot works on my Personal Instagram, but when testing on my Business Instagram it gives this error.
> 
> Error I have: WARNING [2022-02-02 16:10:28] [MYACCOUNT] Unavailable Page: b'LINK TO INSTAGRAM POST' INFO [2022-02-02 16:10:28] [MYACCOUNT] --> Image not liked: b'Unavailable Page'
> 
> Please help? I am stuck.

Check if you can Access your account by app or web, this occurs to me when bot make too much Request to server. It's like a soft Block, After some time it returns to normal.

So avoid make too much request, use supervisor

**Comment by martini09617** [comment]
I am able to login. I tried on 1 other business account and a personal account, and those work. I starting to get crazy!

**Comment by Ticunas** [comment]
> I am able to login as normal. But still the bot isn't working. I tried on 1 other business account and another personal account, and those work. I am starting to get crazy!

When happened to me I as able to login as normal, but when I tried navigate appears message 'Unavailable Page'

**Comment by martini09617** [comment]
#6485  for the full error. @ErwinKoekoek can you help?

**Comment by Prince0Fsaiyan** [comment]
Thank you Erwin! Is there a server to deploy this bot freely so I don't have to keep my mac up and running?

**Comment by KoSv** [comment]
> 1. Is there a server to deploy this bot freely so I don't have to keep my mac up and running? I tried using pythonanywhere because I can deploy my twitter bot easily on that server but with instapy, the server runs into issue after issue. The main issue being: not able to run the firefox webdriver.

use a raspi.

**Comment by Prince0Fsaiyan** [comment]
I just received this error and wanted to bring it to your attention. Not sure of IG changed things on their end or if I've been excessively using the bot bot the error is:
"Image was NOT liked! You have a BLOCK on likes!"
I other posts online saying this is a temporary block and it should recover in a few hours.

**Comment by ErwinKoekoek** [comment]
> I just received this error and wanted to bring it to your attention. Not sure of IG changed things on their end or if I've been excessively using the bot bot the error is: "Image was NOT liked! You have a BLOCK on likes!" I other posts online saying this is a temporary block and it should recover in a few hours.

Hi. This sometimes happens and the bot will continue doing his job. Everything is fine ;-)

**Comment by Prince0Fsaiyan** [comment]
Good morning Erwin. Just another day having fun with your awesome instapy module! I'm trying to deploy a  follower bot for a specific users followers. With every attempted follow, the firefox page shows the following instagram error message:
Try Again Later
We restrict certain activity to protect our community.

I believe this is because Instagram has pointed out my account as "spam" or suspicious. Any tips?

**Comment by workproduct** [comment]
Ban is from 2 hours to 2 weeks depending on severity

**Comment by CyrilDesch** [comment]
There is a problem with business profile, but @igor-zukowski found this solution in util.py.
https://github.com/InstaPy/InstaPy/issues/6421#issuecomment-1030687915
```
def get_additional_data(browser):
    """
    Get additional data object from page source
    Idea and Code by alokkumarsbg

    :param browser: The selenium webdriver instance
    :return additional_data: Json data from window.__additionalData extracted from page source
    """
    additional_data = None
    soup = BeautifulSoup(browser.page_source, "html.parser")
    for text in soup(text=re.compile(r"window.__additionalDataLoaded")):
        if re.search("^window.__additionalDataLoaded", text):
            additional_data = json.loads(re.search("{.*}", text).group())
            return additional_data

    for text in soup(text=re.compile(r"window._sharedData")):
        if re.search("^window._sharedData", text):
            additional_data = json.loads(re.search("{.*}", text).group())
            data = additional_data["entry_data"]["PostPage"][0]
            return data

    return additional_data
```
Everything works perfectly with this repo and this solution.

**Comment by ErwinKoekoek** [comment]
> > @ErwinKoekoek Got it, only the "like_util.py" is still working properly. I am getting an error mentioned here. #6233
> 
> > 
> 
> > More specific, the bot works on my Personal Instagram, but when testing on my Business Instagram it gives this error.
> 
> > 
> 
> > Error I have: WARNING [2022-02-02 16:10:28] [MYACCOUNT] Unavailable Page: b'LINK TO INSTAGRAM POST' INFO [2022-02-02 16:10:28] [MYACCOUNT] --> Image not liked: b'Unavailable Page'
> 
> > 
> 
> > Please help? I am stuck.
> 
> 
> 
> Check if you can Access your account by app or web, this occurs to me when bot make too much Request to server. It's like a soft Block, After some time it returns to normal.
> 
> 
> 
> So avoid make too much request, use supervisor

Hi. You are mentioned as reviewer for merging this PR. Please push the button 👌

**Comment by Wolkex3** [comment]
> > Same Error on Win 10 also I have done #6467 corrections also.
> > Please help!!!
> 
> #6467 did the fix for me.
> 
> 1. Make sure to backup the following files:
> 
> * like_util.py
> * util.py
> * comment_util.py
> 
> On linux these files are stored in : /usr/local/lib/python3/dist-packages/instapy
> 
> 2. Then clear the contents of the original files (like_util.py , util.py and comment_util.py)
> 3. simply copy-paste the entire util.py, comment_util.py & like_util.py content from the link below to your files
>    https://github.com/ErwinKoekoek/InstaPy/tree/master/instapy
> 4. Instapy should be working again.

Hi, I used this code and it finally works. Only error Im getting is following: 
`--> Couldn't follow 'XYZ'! ~user is inaccessible`

Liking is working

**Comment by maury37** [comment]
Agreed with @Wolkex3
- liking is working
- commenting is working. 

But still issues with 
- Business Profiles 
- Following...

**Comment by RainbowTob** [comment]
Same

**Comment by Wolkex3** [comment]
Liking had worked for me with the update of the three files yesterday. But now no longer. I get the following error message. 

@maury37, @RainbowTob - same for you? Or did I break something on my end?

```
WARNING [2022-02-21 21:10:41] [X]  Unavailable Page: b'https://www.instagram.com/p/CaQJFVHvXXX/'
INFO [2022-02-21 21:10:41] [X]  --> Image not liked: b'Unavailable Page'
```

**Comment by maury37** [comment]
> Liking had worked for me with the update of the three files yesterday. But now no longer. I get the following error message.
> 
> @maury37, @RainbowTob - same for you? Or did I break something on my end?
> 
> ```
> WARNING [2022-02-21 21:10:41] [X]  Unavailable Page: b'https://www.instagram.com/p/CaQJFVHvXXX/'
> INFO [2022-02-21 21:10:41] [X]  --> Image not liked: b'Unavailable Page'
> ```

==> Likes still working on my side right now !

**Comment by RainbowTob** [comment]
Like is working. Win10 Instapy 0.6.16

**Comment by PhineasFisher** [comment]
> I can't fix the issue. Do you guys have an idea?

You are soft-blocked by instagram due to too many request. Try again in 24h.

**Comment by Wolkex3** [comment]
> 

I have this issue since 6 days. :(

**Comment by Wolkex3** [comment]
> > I can't fix the issue. Do you guys have an idea?
> 
> You are soft-blocked by instagram due to too many request. Try again in 24h.

Hi @PhineasFisher, you are right. I blocked. I checked it with another account. How can i prevent a softblock next time? I just used the quicktemplates a little bit.

**Comment by maury37** [comment]
Hi @Wolkex3, 
It happens to be blocked, you need to configure session.set_quota_supervisor. It allows your Instapy to stop activities for a period to avoid too many actions. 

Other point :
Following is still not working & business profiles are skipped. Anyone was able to fix it ?

**Comment by Wolkex3** [comment]
> Hi @Wolkex3, 
> 
> It happens to be blocked, you need to configure session.set_quota_supervisor. It allows your Instapy to stop activities for a period to avoid too many actions. 
> 
> 
> 
> 


Hi, is it possible to fix/ update the helper app? And how long will I be blocked. I think it is almost a week already.

**Comment by Asdrubale2022** [comment]
Did all the changes but getting:
Cannot find is_private so defined as None
This user is private and we are not following. 'None':'Following'
But that's not true, the user's image list is available since it's profile is public.

**Comment by skoenig** [comment]
@ErwinKoekoek why is this PR closed? The changes are not in the master branch yet and I think it might get overlooked by the reviewers if it is closed already.

**Comment by ErwinKoekoek** [comment]
> @ErwinKoekoek why is this PR closed? The changes are not in the master branch yet and I think it might get overlooked by the reviewers if it is closed already.

Hi, Because nobody is reviewing it and want to merge it to the master... 
Do after weeks and helping a lot of people I'm done with it.
Best
Erwin

**Comment by martini09617** [comment]
I had this problem for weeks and all of a sudden. It was working again. No idea why?

**Comment by Wolkex3** [comment]
Im blocked since 1 week. How long does it take to be unblocked? I'm still getting this error:

```WARNING [2022-02-21 21:10:41] [X]  Unavailable Page: b'https://www.instagram.com/p/CaQJFVHvXXX/'
INFO [2022-02-21 21:10:41] [X]  --> Image not liked: b'Unavailable Page'```

**Comment by Asdrubale2022** [comment]
@ErwinKoekoek bro can you please send me the files by email? I really appreciate your work, ty. 
asdrubale.rossi@protonmail.com

**Comment by dimitri-ognibene** [comment]
> Im blocked since 1 week. How long does it take to be unblocked? I'm still getting this error:
> 
> ```
> INFO [2022-02-21 21:10:41] [X]  --> Image not liked: b'Unavailable Page'```
> ```

how do you know you are blocked and it is not a problem with the code?

**Comment by Wolkex3** [comment]
> > Im blocked since 1 week. How long does it take to be unblocked? I'm still getting this error:
> 
> > 
> 
> > ```
> 
> > INFO [2022-02-21 21:10:41] [X]  --> Image not liked: b'Unavailable Page'```
> 
> > ```
> 
> 
> 
> @Wolkesx3  how do you know you are blocked and it is not a problem with the code?

@dimitri-ognibene I tested the coffee with a test account. Test was just for a short timeframe. 

I found this in my log. Is this an issue? Maybe the root cause for my problems? Hide Selenium is ok. 

INFO [2022-03-07 21:50:25] [XXXx] - window.navigator.webdriver response: None

**Review by lucas-freitas00 — COMMENTED** [review]
ok

**Pull request closed without merging**